### PR TITLE
Fan-In implementation 

### PR DESCRIPTION
**Pull request by kevinschwarz**
Issue #, if available: No issue open 

Description of changes: Fan-in Implementation

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/hugo/content/en-us/implementations/aws/fan-in/fan-in1/_index.md [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+1. The _AWS IoT Core Message Broker_ has [service quotas](https://docs.aws.amazon.com/general/latest/gr/iot-core.html#message-broker-limits) around number of connect requests per second per account (500 - adjustable), per client (1 - not adjustable) as well as maximum inbound concurrent client connections, publish requests, subscription per connection and more.  
+1. _AWS IoT Rules Engine_ has [service quotas](https://docs.aws.amazon.com/general/latest/gr/iot-core.html#rules-limits) that govern rule invocations per second (20,000 - adjustable), max actions per rule (10 - not adjustable) and more.
+1. Review the _Amazon Kinesis Data Firehose_ [service quotas](https://docs.aws.amazon.com/firehose/latest/dev/limits.html). Be aware that each delivery stream stores data records for up to 24, the _Lambda_ buffer size is between 1 MiB and 3 MiB, and the buffer interval ranges from 60 seconds to 900 seconds.
```
Comment (gadams999) [inline src/hugo/content/en-us/implementations/aws/fan-in/fan-in1/_index.md:271]: up to 24... hours?

**Review by gadams999 — CHANGES_REQUESTED** [review]
Great content! Just a minor structural change and a few comments on word case. I've set to "request changes". If you could review and close out, we'll get this published!

File: src/hugo/content/en-us/implementations/aws/_index.md [hunk 2]
```diff
@@ -14,3 +14,9 @@ This page contains the complete list of implementations per-pattern. If the list
 This is list of AWS implementations for the _Command_ pattern.
 
 {{% children  page="Command"  style="div" depth="2" startdepth="1" description="true" showhidden="true" %}}
+
+## Fan-in
+
+This is list of AWS implemenations for the _Fan-in_ pattern.
+
+{{% children page="Fan-in" style="div" depth="2" startdepth="1" description="true" showhidden="true" %}}
```
[the remark this answers is not in the document]
Comment (kevinschwarz) [reply src/hugo/content/en-us/implementations/aws/_index.md:22]: corrected in next commit

File: src/hugo/content/en-us/implementations/aws/fan-in/_index.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+---
```
[the remark this answers is not in the document]
Comment (kevinschwarz) [reply src/hugo/content/en-us/implementations/aws/fan-in/_index.md:1]: corrected in next commit

File: src/hugo/content/en-us/implementations/aws/fan-in/fan-in1/_index.md [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+- _Devices_ are the IoT things transmitting telemetry
+- _AWS IoT Core_ is the MQTT message broker processing messages on behalf of the clients
+- _IoT Rule_ performs the actions taken on the message
+- _Kinesis_ is the Amazon Kinesis Data Firehose stream for processing and delivering messages to a storage sink
+- _Lambda_ runs your code to act on the incoming messages and transform or enrich them prior to storage
+- _S3_ is the Data Lake where data will land for storage and further analysis or ETL processing
+
+
+1. _Devices_ establish an MQTT connection to the _AWS IoT Core_ endpoint, and then publish message to the `dt/plant1/dev_n/temp` (data telemetry) topic. This is a location and device specific topic to deliver telemetry messages for a given device or sensor.
+1. The _IoT Rule_ queries a wildcard topic `dt/plant1/+/temp` from the _AWS IoT Core_ to consolidate messages across devices for plant1 and PUTs those messages onto a _Amazon Kinesis Data Firehose_ stream.
```
[the remark this answers is not in the document]
Comment (kevinschwarz) [reply src/hugo/content/en-us/implementations/aws/fan-in/fan-in1/_index.md:41]: all occurances updated to match variable naming and avoid english vs. code decrepancies

File: src/hugo/content/en-us/implementations/aws/fan-in/fan-in1/_index.md [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+
+
+@enduml
+```
+
+{{% /center %}}
+
+### Assumptions
+
+This implementation approach assumes all _Devices_ are not connected to the internet or _AWS Iot Core_ at all times. Each _Device_ publishes temperature telemetry to a single topic. The implementations also assumes that all temperature readings are emitted with a sensor name value of Temperature celsius or Temperature Fahrenheit and follow the message payload formats outlined below. This implementation also assumes a very simple flow for the sake of making the example easy to digest. Alternative options are also called out within the sections below.
```
[the remark this answers is not in the document]
Comment (kevinschwarz) [reply src/hugo/content/en-us/implementations/aws/fan-in/fan-in1/_index.md:91]: removed sentence as it did not add to the implementation

File: src/hugo/content/en-us/implementations/aws/fan-in/fan-in1/_index.md [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+{{% /notice %}}
+
+### Devices
+
+Once connected to _AWS IoT Core_, devices will transmit telemetry data to plant and device specific MQTT topics. The below example demonstrates MQTT topics and payloads for device 1 and 2 in plant 1. Your implementation might support hundreds, thousands, or millions of devices. Copy and paste the below topic names and message payloads into the subscribe and publish inputs of the MQTT test client to simulate the device traffic we will fan-in. You can subscribe to each topic to view the messages device specific as you publish them or you can subscribe to the wildcard topic `dt/plant1/+/temp` to see messages in aggregate.
+
+{{< tabs groupId="MQTT-scenario">}}
```
[the remark this answers is not in the document]
Comment (kevinschwarz) [reply src/hugo/content/en-us/implementations/aws/fan-in/fan-in1/_index.md:105]: nice suggestion on device tabs, done

File: src/hugo/content/en-us/implementations/aws/fan-in/fan-in1/_index.md [hunk 7]
```diff
@@ -0,0 +1,273 @@
...
+{{% tab name="MQTT" %}}
+
+MQTT Topic name for the temperature of Device 1 in Plant 1. 
+```yaml
+dt/plant1/dev_1/temp
+```
+Temperature in celsius telemetry JSON payload for Device 1 in Plant 1.
```
[the remark this answers is not in the document]
Comment (kevinschwarz) [reply src/hugo/content/en-us/implementations/aws/fan-in/fan-in1/_index.md:112]: globally corrected

File: src/hugo/content/en-us/implementations/aws/fan-in/fan-in1/_index.md [hunk 8]
```diff
@@ -0,0 +1,273 @@
...
+1. Choose `Enabled` for Transform source records with AWS Lambda and select the function created above `fan-in_device_temperature_converter`
+1. Leave Lambda function version as `$LATEST`
+1. Leave Record format conversion `Disabled` and click `Next`
+1. Leave your Destination as `Amazon S3`
+1. Under S3 Destination choose the bucket you created above `fan-in-telemetry-<AccountId>`
+1. Scroll down and click `Next`
+1. On the Configure settings screen scroll down and click `Next`
+1. On the Review screen click `Create delivery screen`
+
+Alternatively your _AWS IoT Rule_ can achieve the Fan-in pattern with actions that send messages to [Amazon Kinesis Data Streams](https://aws.amazon.com/kinesis/data-streams/) where other applications can read and act on buffered telemetry data, [Amazon Kinesis Data Analytics](https://aws.amazon.com/kinesis/data-analytics/) to perform analytics processing of data real time with SQL or Apache Flink, SQS message queues where you can asynchronously process messages, or any other of of the supported [AWS IoT Rule Actions](https://docs.aws.amazon.com/iot/latest/developerguide/iot-rule-actions.html).
```
[the remark this answers is not in the document]
Comment (kevinschwarz) [reply src/hugo/content/en-us/implementations/aws/fan-in/fan-in1/_index.md:240]: kinesis data stream is the first alternative to KDF (used in this implementation) starting off the list of options.

File: src/hugo/content/en-us/implementations/aws/fan-in/fan-in1/_index.md (hunk 1 above)
Comment (kevinschwarz) [reply src/hugo/content/en-us/implementations/aws/fan-in/fan-in1/_index.md:271]: thank you for catching this

**Review by gadams999 — APPROVED** [review]
Review completed. Will wait for one more requested change out-of-band then merge.

**Pull request merged**